scripts/jenkins/workers/roles/jenkins_worker/defaults/main.yml

```diff
@@ -68,10 +68,13 @@ jenkins_worker_swarm_config:
     regexp: '^LABELS='
   - line: 'PARAMS="-username {{ _swarm_username }} -password {{ _swarm_password }} -master {{ jenkins_worker_master }} -mode exclusive -executors {{ jenkins_workers_executors }} -name {{ ansible_hostname }} -description {{ os_cloud }}/{{ os_project_name }} -disableSslVerification -disableClientsUniqueId -fsroot /home/jenkins/workspace"'
     regexp: '^PARAMS='
-  - line: 'JENKINS_USER="jenkins"'
-    regexp: '^JENKINS_USER='
-  - line: 'JENKINS_HOME="/home/jenkins"'
-    regexp: '^JENKINS_HOME='
+
+jenkins_worker_swarm_systemd_file: "/usr/lib/systemd/system/jenkins-swarm-client.service"
```

Comment:
That might work, but it would be better to use a drop in like /etc/systemd/system/jenkins-swarm-client.service.d/settings.conf so as to not touch an rpm owned file. And to create it if it doesn't exist.

Reply:
That would be nice indeed, however I am afraid that might not work because `User` and `WorkingDirectory` are hardcoded at the systemd file (https://build.opensuse.org/package/view_file/devel:tools:building/jenkins-swarm-client/jenkins-swarm-client.service?expand=1).

If I set `User` and `WorkingDirectory` in `/etc/systemd/system/jenkins-swarm-client.service.d/settings.conf` will it take precedence over the values in the systemd file?

Reply:
That is what should happen. I had a case with Type where it didn't work on an old systemd version, but making a copy to overwrite the whole unit worked. (Here the copy would be to /etc/systemd/system/jenkins-swarm-client.service .) So, would need to test. Lets do that later.